File: clients/typescript/src/model/debug-data.ts
import { Stream } from "../stream";

/**
 * TODO - Document
 */
export class DebugData {

    constructor() {
    }

    /**
     * Read DebugData from input stream
     */
    static async readFrom(stream: Stream): Promise<DebugData> {
        return new DebugData()
    }

    /**
     * Write DebugData to output stream
     */
    async writeTo(stream: Stream) {
    }
}

File: clients/python/model/debug_data.py
from stream_wrapper import StreamWrapper

class DebugData:
    """TODO - Document"""

    __slots__ = ()


    def __init__(self):
        pass

    @staticmethod
    def read_from(stream: StreamWrapper) -> "DebugData":
        """Read DebugData from input stream
        """
        return DebugData()
    
    def write_to(self, stream: StreamWrapper):
        """Write DebugData to output stream
        """
        pass
    
    def __repr__(self):
        return "DebugData(" + \
            ")"